flink-state-backends/flink-statebackend-rocksdb/src/main/java/org/apache/flink/contrib/streaming/state/RocksDBConfigurableOptions.java

```diff
@@ -139,6 +149,25 @@ public class RocksDBConfigurableOptions implements Serializable {
                                     NONE.name(),
                                     LEVEL.name()));
 
+    public static final ConfigOption<CompressionType> COMPRESSION_TYPE =
```

Comment:
Introduce CompressionPerLevel  requires passing in a CompressType List for each level.
I idea is to introduce a config like 
`ConfigOption<List<CompressType>> CompressionPerLevel `
But I'm worried about whether the cost to users is too high. After all, many flink users may not understand rocksdb or even lsm.
Do you have any suggestions ?

Reply:
+1 to the idea of @pnowojski

Reply:
@pnowojski @masteryhx thanks for suggestion . After I read the related documentation and code. I think maybe we can just use the API `public ColumnFamilyOptions setCompressionPerLevel(
      final List<CompressionType> compressionLevels)`  directly. We can pass in a List of CompressionType to represent different levels of Compression Style.


https://github.com/facebook/rocksdb/blob/65e162bf09e6d81d5de533539753d80c008717fd/include/rocksdb/advanced_options.h#L399C1-L426C5
>   // NOTE: if the vector size is smaller than the level number, the undefined
>   // lower level uses the last option in the vector, for example, for 3 level
>   // LSM tree the following settings are the same:
>   // {kNoCompression, kSnappyCompression}
>   // {kNoCompression, kSnappyCompression, kSnappyCompression}

It can be seen that the effect of using this parameter directly is actually the same as @pnowojski 's idea. So it seems that we do not need to rename parameter name to `rocksdb.compression.types` but directly use the following parameters `state.backend.rocksdb.compression.per.level`.
I have updated the pr. Please comment if you have any other suggestions.

Reply:
👍 Keeping it consistent with the rocksdb naming makes sense. Good that someone from RocksDB had the same idea as me how to configure it 😆